Fix SCAN with match

maiha opened the pull request:
It seems SCAN with match doesn't work correctly.

#### scan(0, "foo*")
- should request: `["SCAN", "0", "MATCH", "foo*"]`

#### scan(0, "foo*", 1024)
- should request: `["SCAN", "0", "MATCH", "foo*", "COUNT", "1024"]`

Fixed it in TDD style commits :)

jmoriau commented:
I just noticed sscan hscan zscan also suffer from this since they are exactly the same.

maiha commented:
OK. I'll fix all *scan commands too.
thanks

jmoriau commented:
Awesome thanks! :+1:

maiha commented:
pushed -f

CHANGES from master
- fixed scan,sscan,hscan,zscan about MATCH and COUNT args
- added tests about it

NOTE in test
- added `array` helper method to convert `RedisValue` to `Array(String)` that is same as existing `sort` method except sorting feature
- `scan` spec assumes that the size of all entries in the testing instance should be < 10M
- gave up "match and count" spec for hscan, zscan who doesn't handle count value strictly

thanks!

stefanwille commented:
Merged, thank you for the fixes!
